imagely.es6

```diff
@@ -165,6 +180,50 @@ class Imagely {
 	}
 
 	/**
+	 * Assuming css, html, and js are the same in source files, render pages with unique json data.
+	 * The hope is
+	 *		1) We can make time gains by not opening and closing phantom every image.
+	 *		2) We can make time gains by caching repeated generated css, js, and html.
+	 *
+	 * @param {Object} phantomjs
+	 * @param {String} html Html to inject into the page.
+	 * @param {Object} page phantomjs page instance.
+	 * @param {Object} json The object we're looping over and setting uniquely per page.
+	 */
+	batch(phantomjs, html, page, json) {
+		console.info(`image ${this.batchIndex} of ${this.batchLength}`);
+
+		if (this.batchIndex < this.batchLength) {
+			html = this.setWindowData(html, JSON.stringify(json[this.batchIndex]));
+			page.setContent(html);
+
+			let batchDestination = this.makeUniqueDestination(this.destination, this.batchIndex);
+			this.renderPage(page, phantomjs, batchDestination, this.options, this.callback);
+			this.batchIndex++;
+
+			setTimeout(() => {
+				this.batch(phantomjs, html, page, json);
```

Comment:
Totally agree.

For some reason had problems putting `batch()` in the page.render callback, put it back and it seems to block the calling until after render (like it should).

The batching is now crazy quick, on my machine ~50ms per image batching over 100 images. The code is also much cleaner. Thanks!